File: fileloader/src/main/java/app/task/fileloader/listener/FileLoaderListener.java
package app.task.fileloader.listener;


import app.task.fileloader.model.FileLoaderDataModel;

public interface FileLoaderListener
{
	void onStart(FileLoaderDataModel fileLoaderDataModel);
	
	void onSuccess(FileLoaderDataModel fileLoaderDataModel);
	
	void onFailure(FileLoaderDataModel fileLoaderDataModel, int statusCode, byte[] errorResponse, Throwable e);
	
	void onRetry(FileLoaderDataModel fileLoaderDataModel, int retryNo);
}

File: fileloader/src/main/java/app/task/fileloader/listener/FileLoaderListener.kt
package app.task.fileloader.listener

import app.task.fileloader.model.FileLoaderDataModel

/*
 * Created by venkatesh on 10-Apr-19.
 */

interface FileLoaderListener {
    fun onStart(fileLoaderDataModel: FileLoaderDataModel)

    fun onSuccess(fileLoaderDataModel: FileLoaderDataModel)

    fun onFailure(
        fileLoaderDataModel: FileLoaderDataModel,
        statusCode: Int,
        errorResponse: ByteArray,
        e: Throwable
    )

    fun onRetry(fileLoaderDataModel: FileLoaderDataModel, retryNo: Int)
}
